packages/modular-scripts/react-scripts/config/webpack.config.js

```diff
@@ -27,9 +27,20 @@ const ForkTsCheckerWebpackPlugin = require('fork-ts-checker-webpack-plugin');
 const postcssNormalize = require('postcss-normalize');
 const isCI = require('is-ci');
 
-const esbuildTargetFactory = process.env.ESBUILD_TARGET_FACTORY
-  ? JSON.parse(process.env.ESBUILD_TARGET_FACTORY)
-  : 'es2015';
+const isApp = process.env.MODULAR_IS_APP === 'true';
+const isEsmView = !isApp;
+
+// If it's an app, set it at ESBUILD_TARGET_FACTORY or default to es2015
+// If it's not an app it's an ESM view, then we need es2020
+const esbuildTargetFactory = isApp
+  ? process.env.ESBUILD_TARGET_FACTORY
+    ? JSON.parse(process.env.ESBUILD_TARGET_FACTORY)
+    : 'es2015'
+  : 'es2020';
+
+const { externalDependencies } = process.env.MODULAR_PACKAGE_DEPS
+  ? JSON.parse(process.env.MODULAR_PACKAGE_DEPS)
```

Comment:
Oh I see. This env var is generated at time of execution and passed in with no expectation of user configuration. 

If that's the case, why doesn't the build script just fetch the external dependencies itself instead of reading it from the env var?

Reply:
> If that's the case, why doesn't the build script just fetch the external dependencies itself instead of reading it from the env var?

Because then we would need to either compile our `getPackageDependencies.ts` into javascript at runtime for `webpack.config.js` to use it, or "downgrade" our code to js.

The real problem here is that `webpack.config.js` is not invoked by us, but by Webpack itself and we have to pass parameters to it via the environment, which is something we inherited from create-react-app.

I'm sure there's a way to solve this in a better way (ie generate a JSON configuration file from our build scripts), and I would like to do that, but that's for another PR.

Reply:
> Is this script run in the same process as the script that generates the configuration? If so we could switch to using process.modular = { packageDependencies, isApp, ... }

I'm not really sure what you mean, but this is not really a script we execute, it's the actual Webpack configuration (that can be a javascript file), executed by Webpack itself. What we do is a bit convoluted:
1. In `build/index.ts` we call all our ts utilities, get the dependencies, set the environment variables and start a new **javascript** process in  via `execa` - the process is contained in `react-scripts/scripts/build.js`
2. `build.js` starts webpack, with `const compiler = webpack(configFactory('production'));`. This is taken directly from CRA.
3. The `webpack` library starts a Webpack process, which in turn executes the configuration at `webpack.config.js`  (this file, which needs to be js since is directly `require`d by Webpack, afaik).

Reply:
Agreed to do this in another PR.